File: server2/src/domain/players/player.py
class PlayerId:
    id: str

    @staticmethod
    def from_sid(sid):
        return PlayerId(id=str(sid))


class Player:

    def __init__(self, sid, username: str, game):
        self.__id = PlayerId.from_sid(sid)
        self.__username = username
        self.__game = game

    @property
    def id(self) -> PlayerId:
        return self.__id

    @property
    def username(self) -> str:
        return self.__username

    @property
    def game(self):
        return self.__game

File: server-koa/src/domain/players/player.ts
import { PlayerId } from './playerId'
import { GameId } from '../games/gameId'
import { Equalable } from '../../utils/equalable'


export class Player implements Equalable<Player> {
    readonly id: PlayerId
    private gameId: GameId

    constructor(
        readonly username: string,
        socketId: string
    ) {
        this.id = PlayerId.fromSocketId(socketId)
    }

    setGameId(gameId: GameId) {
        if (this.gameId) {
            throw new Error("this player is already assigned to a game!")
        }

        this.gameId = gameId
    }

    equals(other: Player): boolean {
        return this.id.equals(other.id)
    }
}
